pkg/platform/packet/packet.go

```diff
@@ -288,6 +294,46 @@ func createTerraformConfigFile(cfg *config, terraformPath string) error {
 	return nil
 }
 
+func validateNodePrivateCIDRs(cfg *config) error {
+	// If both the deprecated and the new fields are provided then error out.
+	if cfg.NodePrivateCIDR != "" && len(cfg.NodePrivateCIDRs) > 0 {
+		//nolint:stylecheck
+		return fmt.Errorf("`node_private_cidr` and `node_private_cidrs` cannot be used together." +
+			" Use `node_private_cidrs` only.")
+	}
+
+	// If just the deprecated field is provided then produce a warning.
+	if cfg.NodePrivateCIDR != "" {
+		fmt.Printf("Warning: `node_private_cidr` is deprecated, use `node_private_cidrs` instead.\n")
```

Comment:
Hmm, we could use warning diagnostic, I would really avoid adding `fmt.Printf` here.

Reply:
But we are not returning from here, merely reporting the warning. So how to we notify user right away without `fmt.Printf` despite the fact that we use `hcl.Diagnostic` or not?

Reply:
We do not print warnings at the moment, but we can have _some_ facility to generate them. Actually printing them can be done later IMO.

Reply:
Then what is the point of even having warning if user cannot see them? And this whole deprecation policy also turns useless.

Reply:
>Then what is the point of even having warning if user cannot see them? And this whole deprecation policy also turns useless.

It divides the task, so someone else can work on it (even in parallel) and gives us time to think about proper solution for warnings, rather than doing some one-off things.

It's also an anti-pattern to make code deep down the stack print something directly to stdout.

Reply:
I will give higher preference to what user will see (basically UX) than how and when we (Lokomotive devs) manage to solve this issue of warning-printing.

Reply:
If I use `pkg/platform/packet` as a library (UX) and I get some random prints, I'd report it as a bug.

Reply:
Yep, but for now nobody is using it as a library. We are not providing any guarantees or contracts for it to used as libraries.

Reply:
Still, it's a sort of UI and stdout is a hidden dependency IMO.